test/functional/c/test_magnetic_symmetry.cpp

```diff
@@ -372,12 +372,16 @@ TEST(MagneticSymmetry, test_spg_get_magnetic_dataset_high_mag_symprec) {
     double tensors[] = {0, 0, -1, 1};
     int num_atoms = 4;
     double symprec = 3.0;
+    SpglibError error;
 
     SpglibMagneticDataset *dataset;
     dataset = spg_get_magnetic_dataset(lattice, positions, types, tensors, 0,
                                        num_atoms, 0, symprec);
     // MSG identification is failed with the too high mag_symprec
-    ASSERT_TRUE(dataset == NULL);
+    EXPECT_EQ(dataset, nullptr);
+
+    error = spg_get_error_code();
+    EXPECT_NE(error, SpglibError::SPGLIB_SUCCESS);
```

Comment:
```suggestion
    auto error = spg_get_error_code();
    EXPECT_EQ(error, SpglibError::SPGERR_SPACEGROUP_SEARCH_FAILED);
```
Not sure what the error code should be, but we should test the cause more narrowly, rather than `EXPECT_NE(SPGLIB_SUCCESS)`

Reply:
Indeed. I've specified the error code `SpglibError::SPGERR_ATOMS_TOO_CLOSE` in this case. But, I'm not this error code is appropriate for this test case.

Reply:
It's ok, that's the purpose of the CI to catch and make sure these are predictable.

Also about using `auto`, it would make the code a bit cleaner.